validate/validate.go

```diff
@@ -66,22 +66,22 @@ type Validator struct {
 }
 
 // NewValidator creates a Validator
-func NewValidator(spec *rspec.Spec, bundlePath string, hostSpecific bool, platform string) Validator {
+func NewValidator(spec *rspec.Spec, bundlePath string, hostSpecific bool, platform string) (Validator, error) {
 	if hostSpecific && platform != runtime.GOOS {
-		platform = runtime.GOOS
+		return Validator{}, fmt.Errorf("When hostSpecific is set, platform must be same as the host platform")
 	}
 	return Validator{
 		spec:         spec,
 		bundlePath:   bundlePath,
 		HostSpecific: hostSpecific,
 		platform:     platform,
-	}
+	}, nil
 }
 
 // NewValidatorFromPath creates a Validator with specified bundle path
 func NewValidatorFromPath(bundlePath string, hostSpecific bool, platform string) (Validator, error) {
 	if hostSpecific && platform != runtime.GOOS {
-		platform = runtime.GOOS
+		return Validator{}, fmt.Errorf("When hostSpecific is set, platform must be same as the host platform")
```

Comment:
I'm fine with this early check, as you have it.  But I'd also be fine dropping this block, and letting `NewValidator` handle the check for us.  Leaving it to `NewValidator would be slightly more DRY (good), but a slightly later check (maybe bad if you validate lots of bundles, or on wimpy hardware?).

Reply:
I think it's better to do it on this side, otherwise the other checks in `NewValidatorFromPath` would be redundant if it went wrong.

Reply:
> … otherwise the other checks in `NewValidatorFromPath` would be redundant if it went wrong…

I'm not clear on this point.  `NewValidator` has exactly the same check as you have here (you are touching both checks in this PR), and `NewValidatorFromPath` calls `NewValidator` internally (you're touching that line below in this PR).  So the choice is either “check for this explicitly in both `NewValidator` and `NewValidatorFromPath`” (which is what you're doing in 65846d77) or “check for this explicitly in `NewValidator`, and have `NewValidatorFromPath` check indirectly via `NewValidator`” (which is what we'd have if we dropped this block from `NewValidatorFromPath`).  As I said, I'm fine with either of those, but checking only in `NewValidator` (and dropping this block here) would be slightly more DRY.

Reply:
Please remove this block

Reply:
updated.